src/components/Web3Provider/index.tsx

```diff
@@ -16,17 +22,19 @@ export default function Web3Provider({ children }: { children: ReactNode }) {
 
   return (
     <Web3ReactProvider connectors={connectors} key={key}>
-      <Tracer />
+      <Updater />
       {children}
     </Web3ReactProvider>
   )
 }
 
-function Tracer() {
-  const { chainId, provider } = useWeb3React()
+/** A component to run hooks under the Web3ReactProvider context. */
+function Updater() {
```

Comment:
nit: easier to read to me if components are defined before they're used. also maybe `ConnectionUpdater` is more clear? looks like it is basically just used for debug/analytics of the connection

Reply:
this deserves to be a lint rule, because my general preference is the opposite.